### PR TITLE
Filter out Breaking changes info and return in automation result

### LINKED ISSUE #46859: [SDK Automation] The breaking change info does not show in the result
When there are breaking changes, SDK automation can detect out the breaking changes, but the content of breaking changes in automation result is empty, see:
Image

expected: 
the breaking change info are included in the automation result, like
Image

### PR DESCRIPTION
**Pull request by chunyu3**
# Contributing to the Azure SDK
Fix https://github.com/Azure/azure-sdk-for-net/issues/46859

- update the filter constrains in the breaking change result. Current it skip first of 2 which cause the breaking change log missing. Update to skip only the latest one which is the `ApiCompat` information
- Add regex to retrieve out the breaking change info from log and return it in the automation result

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: eng/scripts/automation/GenerateAndBuildLib.ps1 [hunk 1]
```diff
@@ -856,8 +857,15 @@ function GeneratePackage()
...
-                $breakingChanges = $logFile -join ",`n"
+                Write-Host "Breaking changes detected in the build log."
+                $logFile = Get-Content -Path $logFilePath | select-object -SkipLast 1
+                $regex = "error( ?):( ?)(?<breakingChange>.*) .*\["
+                foreach ($line in $logFile) {
+                    if ($line -match $regex) {
+                        $breakingChangeItems += $matches["breakingChange"]
+                    }
+                }
+                $breakingChanges = $breakingChangeItems -join ",`n"
```
Comment (raych1) [inline eng/scripts/automation/GenerateAndBuildLib.ps1:868]: Is `$breakingChanges` still useful?

Comment (chunyu3) [reply eng/scripts/automation/GenerateAndBuildLib.ps1:868]: yes. it will be used in `content`

**Comment by raych1** [comment]
@chunyu3  
are you going to use this PR to resolve this issue either?
https://github.com/Azure/azure-sdk-tools/issues/9221

**Comment by chunyu3** [comment]
> @chunyu3 are you going to use this PR to resolve this issue either? [Azure/azure-sdk-tools#9221](https://github.com/Azure/azure-sdk-tools/issues/9221)

hello @raych1 , no ,I will resolve it in another PR.

**Review by raych1 — APPROVED** [review]

**Pull request merged**